pkg/machine/proxyenv/env.go

```diff
@@ -24,7 +24,7 @@ rm -f $SYSTEMD_CONF $ENVD_CONF $PROFILE_CONF
 
 echo "[Manager]" >> $SYSTEMD_CONF
 for proxy in %s; do
-	printf "DefaultEnvironment=%%q\n" "$proxy"  >> $SYSTEMD_CONF
+	printf "DefaultEnvironment=\"%%s\"\n" "$proxy"  >> $SYSTEMD_CONF
```

Comment:
This will break if the vars contains a quote, I guess I am fine to trade quote is broken vs command is broken because the later is actually used in these vars while the quote most likely is never used.

However please fix the test and add a comma there
https://github.com/containers/podman/blob/77f5b4d67867aa76676442b74b5621e21885038e/pkg/machine/e2e/proxy_test.go#L79